File: python/Komponen.py
class Komponen:
    def __init__(self, merk="", nama=""):
        self.merk = merk
        self.nama = nama
    
    def set_merk(self, merk):
        self.merk = merk
    
    def set_nama(self, nama):
        self.nama = nama
    
    def get_merk(self):
        return self.merk
    
    def get_nama(self):
        return self.nama
    
    def display_info(self):
        print(f"Merk: {self.merk}, Nama: {self.nama}", end="")

File: cpp/Komponen.cpp
#ifndef KOMPONEN_CPP
#define KOMPONEN_CPP

#include <iostream>
#include <string>

using namespace std;

class Komponen {
private:
    string merk;
    string nama;
public:
    Komponen() {}

    Komponen(string merk, string nama) {
        this->merk = merk;
        this->nama = nama;
    }

    void setMerk(string merk) {
        this->merk = merk;
    }

    void setNama(string nama) {
        this->nama = nama;
    }

    string getMerk() {
        return merk;
    }
    
    string getNama() {
        return nama;
    }

    virtual void displayInfo() {
        cout << "Merk: " << merk << ", Nama: " << nama;
    }

    ~Komponen() {}
};

#endif // KOMPONEN_CPP
